lib/models/profile.ts

```diff
@@ -51,24 +70,56 @@ export class ProfileModel {
           name: "@fiscalCode",
           value: fiscalCode,
         }],
-        query: "SELECT * FROM profiles p WHERE (p.fiscalCode = @fiscalCode)",
+        query: "SELECT * FROM profiles p WHERE (p.fiscalCode = @fiscalCode) ORDER BY version DESC",
       },
     );
   }
 
   /**
-   * Upserts a Profile
-   *
-   * TODO: add versioning
+   * Create a new Profile
    *
    * @param profile The new Profile object
    */
-  public createOrUpdateProfile(profile: INewProfile): Promise<IRetrievedProfile | null> {
+  public createProfile(profile: IProfile): Promise<IRetrievedProfile> {
+    return new Promise((resolve, reject) => {
+      // the first version of a profile is 0
+      const initialVersion = 0;
+      // the ID of each profile version is composed of the profile ID and its version
+      // this makes it possible to detect conflicting updates (concurrent creation of
+      // profiles with the same profile ID and version)
+      const profileId = generateVersionedProfileId(profile.fiscalCode, initialVersion);
+      DocumentDbUtils.createDocument(
```

Comment:
maybe we can just `return  DocumentDbUtils.createDocument()` without promise wrapping